src/transports/utils/add_query_string.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * Add to an URL a query string corresponding to `supplementaryQueryStringData`,
+ * being tuples where the first element is a query string's property name and the
+ * second element is its value (or null if there's no associated value.
+ *
+ * If the given URL already has a query string, the new query string elements
+ * will just be appended.
+ * @param {string} baseUrl
+ * @param {Array.<Array.<string|null>>} supplementaryQueryStringData
+ * @returns {string}
+ */
+export default function addQueryString(
+  baseUrl: string,
+  supplementaryQueryStringData: Array<[string, string | null]>,
+): string {
+  if (supplementaryQueryStringData.length === 0) {
+    return baseUrl;
+  }
+  let urlPrefix: string;
+  let urlSuffix: string | null;
+  let queryStringStartingChar: "?" | "&" | "";
+  const indexOfQueryString = baseUrl.indexOf("?");
```

Comment:
There is an edge case that seems to not be handled, there can be `?` within fragments

```js
addQueryString("https://www.example.com#questionmark?hello", [
      ["first", "val"],
    ])
// expected   https://www.example.com?first=val#questionmark?hello
// result     https://www.example.com#questionmark?hello&first=val

```

Reply:
:D you're right, I'll also add the test case

Reply:
Done